build/Jenkinsfile.pr

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env groovy
+
+@Library('shared-pipelines') _
+import org.mattermost.Utils
+import java.util.*;
+def utils = new org.mattermost.Utils()
+def rnd = UUID.randomUUID().toString()
+
+pipeline {
+    agent {
+        label 'default-mm-builder'
+    }
+
+    options { buildDiscarder(logRotator(numToKeepStr: '3')) }
+
+    environment {
+        DCNAME = "${rnd}"
+    }
+
+    stages {
+        stage('Stop Old Builds') {
+            steps {
+                script {
+                    utils.stopOldBuilds()
+                }
+            }
+        }
+
+        stage('Checkout'){
+            steps {
+                cleanWs notFailBuild: true
+                sh """
+                    mkdir -p src/github.com/mattermost/mattermost-server
+                    mkdir -p src/github.com/mattermost/mattermost-webapp
+                    mkdir -p src/github.com/mattermost/enterprise
+                """
+                dir('src/github.com/mattermost/mattermost-server') {
+                    checkout scm
+                }
+                dir('src/github.com/mattermost/mattermost-webapp') {
+                    checkout changelog: false, poll: false, scm: [$class: 'GitSCM', branches: [[name: '*/master']], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: '310159d3-f7c5-4f5d-bfa1-151e3ef2db57', url: 'https://github.com/mattermost/mattermost-webapp.git']]]
+                }
+                dir('src/github.com/mattermost/enterprise') {
+                    checkout changelog: false, poll: false, scm: [$class: 'GitSCM', branches: [[name: '*/master']], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: '310159d3-f7c5-4f5d-bfa1-151e3ef2db57', url: 'https://github.com/mattermost/enterprise.git']]]
+                }
+            }
+        }
+
+        stage('Get Enterprise/Webapp') {
+            steps {
+                dir('src/github.com/mattermost/enterprise') {
+                    ansiColor('xterm') {
+                        sh """
+                            #!/bin/bash -ex
+                            git checkout $env.BRANCH_NAME || git checkout $env.CHANGE_BRANCH || git checkout $env.GIT_BRANCH || git checkout master || echo 1
+                            export EE_GIT_COMMIT=\$(git rev-parse HEAD)
+
+                            echo EE Commit: \${EE_GIT_COMMIT}
+                        """
+                    }
+                }
+                dir('src/github.com/mattermost/mattermost-webapp') {
+                    withDockerContainer(args: '', image: 'mattermost/mattermost-build-webapp:v5.2.0') {
```

Comment:
Might be nice not to tie the docker images to versions. Just have an incrementing number when we need to make changes.

Reply:
ok will build based on a date like 20181204-1

Reply:
sure.

Reply:
follow the same in the webapp